### PR TITLE
Enhance TemplateVarShop Function with Shop Group ID for Improved Multi-Shop Support

### PR DESCRIPTION
**Pull request by OrionCodeDev**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Added the parameter 'id_shop_group' => $this->context->shop->id_shop_group to the function getTemplateVarShop(). This change allows better utilization and differentiation of multiple shop groups in templates. <br> This modification affects the template variables for shops by including the shop group ID, facilitating improved customization and handling in multi-shop environments.
| Type?             | improvement
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Verify that the getTemplateVarShop() function now includes the 'id_shop_group' parameter in its output. Ensure templates utilizing this function correctly display and differentiate shop groups. Test with multiple shop groups to confirm the functionality. Now you can use in FO for example {if $shop.id_shop_group == 3}
| UI Tests          | Please run UI tests and paste here the link to the run. As we support MySQL and MariaDB in our tests, you have to launch 2 campaigns (by choosing both). [Read this page to know why and how to use this tool](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/ui-tests/).
| Fixed issue or discussion?     | --
| Related PRs       | --
| Sponsor company   | Code Olympus Studio

**I am unsure which branch this change should be merged into, as this improvement would benefit all versions of PrestaShop. Ideally, this change should be included in every version, as it enhances the functionality by allowing better differentiation and utilization of multiple shop groups in templates.**

### CODE REVIEW

**Comment by ps-jarvis** [comment]
Hi, thanks for this contribution!

I found some issues with the Pull Request description:

* The `branch` should be `develop` or `8.1.x`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))

Would you mind having a look at it? This will help us understand how interesting your contribution is, thank you very much!

(Note: this is an automated message, but answering it will reach a real human)

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -1698,6 +1698,7 @@ public function getTemplateVarShop()
 
         $shop = [
             'id' => $this->context->shop->id,
+            'id_shop_group' => $this->context->shop->id_shop_group,
```
Comment (kpodemski) [inline classes/controller/FrontController.php]:
```suggestion
            'group_id' => $this->context->shop->id_shop_group,
```

Just a suggestion, to me`$shop.id` and `$shop.group_id` look more consistent.

**Comment by kpodemski** [comment]
Thanks for your PR @ChronoBrake 

> I am unsure which branch this change should be merged into, as this improvement would benefit all versions of PrestaShop. Ideally, this change should be included in every version, as it enhances the functionality by allowing better differentiation and utilization of multiple shop groups in templates.

We tend to allow new changes in the develop branch, which means they will be part of the next major or minor version. Following release schedule the next version will be PrestaShop 9, **but!**, I think your change is a nice and safe addition that could be added to PrestaShop 8.1.x branch as well. That means that if you target `8.1.x` branch, your change could be added to PrestaShop 8.1.8 and PrestaShop 9 :)

**Comment by OrionCodeDev** [comment]
> Thanks for your PR @ChronoBrake
> 
> > I am unsure which branch this change should be merged into, as this improvement would benefit all versions of PrestaShop. Ideally, this change should be included in every version, as it enhances the functionality by allowing better differentiation and utilization of multiple shop groups in templates.
> 
> We tend to allow new changes in the develop branch, which means they will be part of the next major or minor version. Following release schedule the next version will be PrestaShop 9, **but!**, I think your change is a nice and safe addition that could be added to PrestaShop 8.1.x branch as well. That means that if you target `8.1.x` branch, your change could be added to PrestaShop 8.1.8 and PrestaShop 9 :)

Thank you for the clarification!

Do I need to take any additional steps to target the 8.1.x branch, or will you handle the necessary adjustments to ensure the change is included in both PrestaShop 8.1.8 and PrestaShop 9?

Thank you for your assistance.

**Comment by kpodemski** [comment]
> > Thanks for your PR @ChronoBrake
> > > I am unsure which branch this change should be merged into, as this improvement would benefit all versions of PrestaShop. Ideally, this change should be included in every version, as it enhances the functionality by allowing better differentiation and utilization of multiple shop groups in templates.
> > 
> > 
> > We tend to allow new changes in the develop branch, which means they will be part of the next major or minor version. Following release schedule the next version will be PrestaShop 9, **but!**, I think your change is a nice and safe addition that could be added to PrestaShop 8.1.x branch as well. That means that if you target `8.1.x` branch, your change could be added to PrestaShop 8.1.8 and PrestaShop 9 :)
> 
> Thank you for the clarification!
> 
> Do I need to take any additional steps to target the 8.1.x branch, or will you handle the necessary adjustments to ensure the change is included in both PrestaShop 8.1.8 and PrestaShop 9?
> 
> Thank you for your assistance.

Yo have two options.

You can change the base ranch and [rebase your pull request against `8.1.x`](https://build.prestashop-project.org/howtos/misc/rebasing-your-git-branch/), or, a bit easier way, to open a new pull request against `8.1.x` branch.

**Review by Hlavtox — APPROVED** [review]

**Comment by OrionCodeDev** [comment]
> > > Thanks for your PR @ChronoBrake
> > > > I am unsure which branch this change should be merged into, as this improvement would benefit all versions of PrestaShop. Ideally, this change should be included in every version, as it enhances the functionality by allowing better differentiation and utilization of multiple shop groups in templates.
> > > 
> > > 
> > > We tend to allow new changes in the develop branch, which means they will be part of the next major or minor version. Following release schedule the next version will be PrestaShop 9, **but!**, I think your change is a nice and safe addition that could be added to PrestaShop 8.1.x branch as well. That means that if you target `8.1.x` branch, your change could be added to PrestaShop 8.1.8 and PrestaShop 9 :)
> > 
> > 
> > Thank you for the clarification!
> > Do I need to take any additional steps to target the 8.1.x branch, or will you handle the necessary adjustments to ensure the change is included in both PrestaShop 8.1.8 and PrestaShop 9?
> > Thank you for your assistance.
> 
> Yo have two options.
> 
> You can change the base ranch and [rebase your pull request against `8.1.x`](https://build.prestashop-project.org/howtos/misc/rebasing-your-git-branch/), or, a bit easier way, to open a new pull request against `8.1.x` branch.

Okay, thanks for help. I will open new pull to 8.1.x
#36451

this one can be closed.

**Pull request closed without merging**